google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -82,7 +84,7 @@ public ServiceRpc create(SpannerOptions options) {
   private SpannerOptions(Builder builder) {
     super(SpannerFactory.class, SpannerRpcFactory.class, builder, new SpannerDefaults());
     numChannels = builder.numChannels;
-    userAgent = builder.userAgentPrefix;
```

Comment:
what happens if the user also specifies this userAgentPrefix?

Reply:
Good catch. That is a dead code now (user can specify userAgentPrefix in the builder, but it will be ignored). I've removed the rest of the related code.

Reply:
Can you give an example of how users would now specify customer user agent with spanner library? Ideally that should be prefixed to the default UA and not replace it completely.

Reply:
It is prefixed. In fact grpc itself does not allow to completely replace User-Agent: https://github.com/grpc/grpc-java/blob/c9b02db276403db4794c6e5ffc78b46889cd4ce8/netty/src/main/java/io/grpc/netty/Utils.java#L113

As for the example: it is exactly same as the examples in the original comment to this PR (the manual clients section). The provided User-Agent will be merged, not completelly replaced. It is also documented in the `ServiceOptions.setHeaderProvider()` method.

Reply:
Is there some common documentation (not Javadoc) somewhere that we can point our customers to on how to set these headers.

Reply:
No. I hope it should not be more difficult than it was before this change. If somebody tries to set header he/she would be searching for some header setter method in the `SpannerOptions.Builder` (same as it was before) and will find `setHeadersProvider()` method (instead of `setUserAgentPrefix()`). It would be slightly less straightforward than before, but I believe should be still ok.